value.go

```diff
@@ -419,34 +424,33 @@ func withHiddenFromBool(with bool) Hidden {
 	return withoutHidden
 }
 
-// Hack - we need to distinguish not-checked-yet and no error situations
-// so we have a special value for no error and nil means that we don't know yet.
-var errNoErrorInObjectInvariants = errors.New("no error - assertions passed")
+func (*valueObject) getType() *valueType {
+	return objectType
+}
 
-type valueObjectBase struct {
-	valueBase
-	assertionError error
+func (o *valueObject) index(i *interpreter, trace TraceElement, field string) (value, error) {
+	return objectIndex(i, trace, objectBinding(o), field)
 }
 
-func (*valueObjectBase) getType() *valueType {
-	return objectType
+func (obj *valueObject) inheritanceSize() int {
```

Comment:
Does this ever get called on the top level (or if it does, do you happen to have the inside object readily available)?

Reply:
It does, but only in context where we are playing with the object internals anyway, so we can reach for the uncachedObject. I removed it.

It comes with an added advantage - valueObject no longer conforms to uncachedObject interface - so they cannot be passed there by mistake (and there were some places where it was unnecessarily passed instead of uncachedObject). It is now fixed. 

Thanks for noticing this.